src/commands/command.ts

```diff
@@ -1,102 +1,96 @@
-   * @param channel - The channel to execute the command on.
-   * @param user - The user trying to execute the command.
+   * @param message - The message triggering the command.
    * @param match - The RegExp match of the command trigger.
    */
-  public async execute(
-    bot: BotClient,
-    message: Message,
-    match: RegExpMatchArray,
-  ): Promise<boolean> {
+  public async execute(message: Message, match: RegExpMatchArray): Promise<boolean> {
     // Check if the user has the required role to execute the command.
+    const bot = message.channel.bot;
+
     if (await message.user.hasRole(message.channel, this.role)) {
-      await this.callback(bot, message, match);
+      await this.action(message, match);
       const time = Date.now() - message.timestamp.valueOf();
-      bot.logger.debug(`Command '${this.label}' executed in ${time} ms.`);
+      if (!(this instanceof CommandGroup)) {
+        bot.logger.debug(`Command '${this.name}' executed in ${time} ms.`);
+      }
       return true;
     }
-    bot.logger.debug(`Command: ${this.label}: ${this.role} role required.`);
+
+    bot.logger.debug(`Command: ${this.name}: ${this.role} role required.`);
     bot.sendMessage(
       message.channel,
       `You need the role '${this.role}' on this server to execute this command!`,
     );
+
     return false;
   }
 
   /** Gets the RegExp used to trigger the command
    *
-   * @param client - The BotChannel to trigger the command on.
+   * @param channel - The channel to get the regex for.
    */
-  public async getRegExp(channel: Channel) {
-    const bot = channel.client;
-    const userTag = EscapeRegex(await bot.getUserTag());
-    const channelPrefix = EscapeRegex(channel.getPrefix());
-    const prefix = this.hasPrefix
-      ? `^\\s*((${userTag})|((${channelPrefix})(\\s*${userTag})?)|((${bot.prefix})\\s*(${userTag})))\\s*`
-      : '';
-
-    const regexString = prefix + this.trigger;
-    return new RegExp(regexString);
+  public getRegExp(channel: Channel): RegExp {
+    const regexString = regexToString(this.channelTrigger(channel));
+    // Allow the bot tag after every command
+    return new RegExp(`${regexString}(\\s*${EscapeRegex(channel.bot.getUserTag())})?\\s*`);
   }
 
-  /** Gets the complete trigger label in the given channel, e.g. '/subscribe'
+  /** Tires to find the label of the given command.
    *
-   * @param channel - The channel to get the trigger label in.
-   * @returns The complete trigger label, e.g. '/subscribe'
+   * @param command - The command to find.
+   * @param channel - The channel to find the label for.
    */
-  public getTriggerLabel(channel: Channel): string {
-    const prefixString = this.hasPrefix ? channel.getPrefix() : '';
-    return prefixString + this.triggerLabel;
-  }
+  public abstract findCmdLabel(command: Command, channel: Channel): string;
+
+  /** Aggregates all commands. */
+  public abstract aggregateCmds(role?: UserRole): Command[];
```

Comment:
this doesn't make a lot of sense to me as part of the `Command` class, what kind of aggregation can you do in a single command instance?

Reply:
I moved it to `CommandGroup`.